### PR TITLE
diff: fix `path_limit` with no change

### PR DESCRIPTION
**Pull request by hugebig**
### Description

fixes https://github.com/inveniosoftware/dictdiffer/issues/172

### Checklist

- [x] I've added relevant test cases.

### CODE REVIEW

**Comment by joesolly** [comment]
This change looks good. Could you update the docs on `path_limit` to describe the change in effect (a diff is still performed beyond the path_limit, but individual differences will be aggregated up to the `path_limit`).

**Review by joesolly — APPROVED** [review]

File: dictdiffer/__init__.py [hunk 1]
```diff
@@ -203,6 +209,9 @@ def check(key):
                 # callees again diff function to compare.
                 # otherwise, the change will be handled as `change` flag.
                 if path_limit and path_limit.path_is_limit(_node + [key]):
+                    if deepcopy(_first[key]) == deepcopy(_second[key]):
```
Comment (jirikuncar) [inline dictdiffer/__init__.py:212]: Do we really need to call `deepcopy` for comparing values?

Comment (hugebig) [reply dictdiffer/__init__.py]: It's not necessary.

**Review by jirikuncar — DISMISSED** [review]

**Review by jirikuncar — APPROVED** [review]

**Comment by Xaelias** [comment]
👋 
Any way this could be merged / released?
I just discovered this bug as well and while I can just skip the diffus for now it wound be nice to not have to 🙂

**Pull request merged**